kernel/api/src/main/java/org/sakaiproject/authz/api/AuthzGroupService.java

```diff
@@ -79,6 +79,14 @@
 	static final String AUTH_ROLE = ".auth";
 
 	/**
+	 * Get all provider IDs for the realms given.
+	 * 
+	 * @param realmIDs a List of the realms you want the provider IDs for.
+	 * @return a Map, where the key is the realm ID, and the value is a List of Strings of provider IDs for that realm
+	 */
+	public Map<String, List<String>> getProviderIDsForRealms( List<String> realmIDs );
```

Comment:
If I remember correctly, we tested several different ways of getting the provider IDs, including `BaseSite.getProviderGroupId()`. We found that we weren't getting the greatest performance out of them, so we decided to write our own to get the IDs in the simplest, quickest manner.